Implement random respawning

nabijaczleweli opened the pull request:
Checks for collisions with bullets and players.
Also moves server stuff to `server` module, cleanliness.

Beef is https://github.com/LoungeCPP/Tatsoryk/compare/b8d3df320b95fc43e1d06a51269cc9db28ea1339...681fa68cf1999a1242c4152e8d96fe3ffd102883, rest is "noise".

TheCatPlusPlus commented on server/src/server/gamestate.rs:
```diff
@@ -183,4 +185,32 @@ impl GameState {
     }
+
+    fn random_free_spot<R: Rng>(&self, rng: &mut R) -> (f32, f32) {
+        loop {
```
Needs some limit on iteration count, so we don't loop endlessly in the worst case.

nabijaczleweli replied:
Done

TheCatPlusPlus commented on server/src/server/gamestate.rs:
```diff
@@ -183,4 +186,38 @@ impl GameState {
+        }
+        println!("Failed to find a random empty spot for player after {} iterations", MAX_ITERATIONS);
+
+        // Hardcoded map boundaries
```
Constant maybe then?

nabijaczleweli replied:
Done

TheCatPlusPlus commented:
:+1:

nabijaczleweli committed:
[Re]spawn player in random free place on the map

Max iterations done to find a viable spot for the player are limited and
each happenstance is logged

Use a static for hardcoded map boundaries

Extract Pythagoras to math module

Move rand to dependencies from dev-dependencies
Move server stuff and game_loop to server module
